go/vt/tabletserver/query_executor.go

```diff
@@ -295,25 +295,38 @@ func (qre *QueryExecutor) execDDL() (*sqltypes.Result, error) {
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "DDL is not understood")
 	}
 
+	if qre.transactionID != 0 {
```

Comment:
I went ahead and added a marker on the TxConnection -- see `ReloadSchemaOnCommit`. Not sure if you have a different approach you'd prefer @sougou